### DISCUSSION TITLE
Smooth scrolling with click on Anchor link

### DISCUSSION THREAD

**Discussion by tianhuat** [quasarframework/quasar#11323] [General - Components / Directives / etc]
Is there a component/utility that helps with smooth scrolling when we click on an anchor link (e.g., https://www.w3schools.com/howto/howto_css_smooth_scroll.asp#section2 ) in Quasar?

**Comment by hawkeye64** [comment]
@tianhuat You mean when you click on a link in a drawer and you want to smooth scroll to an anchor link on the visible page?

In your HTML (template). Presumably on a QItem, when iterating through an array of items and `item.id` is the anchor link ID:

```html
@click="scrollTo(item.id)
```

```js
// your imports
import { scroll } from 'quasar'
const { setVerticalScrollPosition } = scroll
```

```js
    function scrollTo (id) {
      const el = document.getElementById(id)

      if (el) {
        scrollPage(el)
      }
    }

    function scrollPage (el) {
      const rect = el.getBoundingClientRect(),
        scrollTop = window.pageYOffset || document.documentElement.scrollTop,
        offset = rect.top + scrollTop - 50 // allow for toolbar height

      setVerticalScrollPosition(window, offset, 500)
    }
    ```

**Comment by rstoenescu** [comment]
```
import { scroll } from 'quasar'
const { animVerticalScrollTo } = scroll
// ...
animVerticalScrollTo (el, to, duration)
```

**Comment by Nerderer** [comment]
You also have a native `scrollIntoView` function in javascript. Should be supported by most browsers.

`element.scrollIntoView({behavior: "smooth"})`
